rtf_vis_tool/src/Components/CoordinateGrid.js

```diff
@@ -0,0 +1,68 @@
+import React from "react";
+import {Line, Html} from "drei";
+
+//Function that creates the grid lines along with xy plane
+const getGridLines = () => {
+    var coordPairSet = []
+    //add x grid lines
+    for (var z = -10; z <= 10; z++) {
+      coordPairSet.push([[-10, 0, z], [10, 0, z]]);
+    }
+    //add z grid lines
+    for (var x = -10; x <= 10; x++) {
+      coordPairSet.push([[x, 0, -10], [x, 0, 10]]);
+    }
+
+    var finalLineGridSet = []
+    coordPairSet.map(coordPair => {
+      finalLineGridSet.push(<Line points={coordPair} color="black" position={[0,0,0]} lineWidth={0.1}/>);
+    });
+    return finalLineGridSet;
+}
+
+//Component which renders the x,y,z axes along with the grid lines along xy plane
+const CoordinateGrid = () => { 
+    return (
+      <mesh>
+          <Line 
+            points={[[-10,0,0], [10,0,0]]} 
```

Comment:
10 and 50 seem like magic numbers -- is there a simple way to avoid that?

Reply:
this was just a stylistic choice to make it render nicely. I can wrap these in variables and document them so it doesn't seem hard coded.